### ISSUE TITLE
Generated result from nested mixins is wrong.

### ISSUE THREAD

**Issue by aGeNET** [leafo/scssphp#423]
I'm using scssphp 0.6.3 and Bootstrap 4 alpha 2. There are special mixins called `media-breakpoint-only` and `media-breakpoint-between` which have other mixins included.

``` scss
@mixin media-breakpoint-between($lower, $upper, $breakpoints: $grid-breakpoints) {
    @include media-breakpoint-up($lower, $breakpoints) {
        @include media-breakpoint-down($upper, $breakpoints) {
            @content;
        }
    }
}
@mixin media-breakpoint-only($name, $breakpoints: $grid-breakpoints) {
    @include media-breakpoint-up($name, $breakpoints) {
        @include media-breakpoint-down($name, $breakpoints) {
            @content;
        }
    }
}
```

My given scss code for the `media-breakpoint-only` variant looks like this:

``` scss
body {
    background-color: #F0F;
    @include media-breakpoint-only(xs) {
        background-color: red;
    }
    @include media-breakpoint-only(sm) {
        background-color: blue;
    }
    @include media-breakpoint-only(md) {
        background-color: green;
    }
    @include media-breakpoint-only(lg) {
        background-color: gold;
    }
    @include media-breakpoint-only(xl) {
        background-color: brown;
    }
}
```

The expected result should look like this (sassmeister.com puts it out correctly).

``` css
body {
    background-color: #F0F;
}
@media (max-width: 543px) {
    body {
        background-color: red;
    }
}
@media (min-width: 544px) and (max-width: 767px) {
    body {
        background-color: blue;
    }
}
@media (min-width: 768px) and (max-width: 991px) {
    body {
        background-color: green;
    }
}
@media (min-width: 992px) and (max-width: 1199px) {
    body {
        background-color: gold;
    }
}
@media (min-width: 1200px) {
    body {
        background-color: brown;
    }
}
```

But the generated result from scssphp is buggy and wrong.

``` css
body {
    background-color: #f0f;
}
@media (max-width: 543px) {
    body {
        background-color: red;
    }
}
@media (min-width: 544px) {
}
@media (min-width: 768px) and (max-width: 767px) {
    body {
        background-color: blue;
    }
}
@media (min-width: 768px) {
}
@media (min-width: 992px) and (max-width: 991px) {
    body {
        background-color: green;
    }
}
@media (min-width: 992px) {
}
@media (min-width: 1200px) and (max-width: 1199px) {
    body {
        background-color: gold;
    }
}
@media (min-width: 1200px) {
    body {
        background-color: brown;
    }
}
```

Similar wrong results will be generated by scssphp for `media-breakpoint-between`.

**Comment by ahukkanen** [comment]
Very true, I added +1 to the original message.

This also seems to crash the PHP process in some occasions (or make it do an infinite loop). Not sure how to replicate this issue but I think by fixing the bug mentioned above, I'm pretty confident it would fix this as well.

This bug was also experienced with the current version of the Bootstrap v4 GitHub version and with the exact same mixins as in the original message.

**Comment by ahukkanen** [comment]
And generally nested mixins work fine but I think this particular issue is somehow related to the nested `@media` queries.

**Comment by robocoder** [comment]
Off-hand, this looks like a duplicate of #409.

**Comment by mahagr** [comment]
Much simpler test case:

```scss
body {
    @media (min-width: $min) {
      @media (max-width: $max) {
        background-color: white;
      }
    }
}
```

Result is:

```css
@media (min-width: ) and (max-width: ) {
  body {
    background-color: white;
  }
}
```

**Comment by mahagr** [comment]
Minimal test case for this:

```scss
@mixin media-breakpoint-up() {
  $min: 544px;
  @media (min-width: $min) {
    @content;
  }
}
@mixin media-breakpoint-down() {
  $max: 768px;
  @media (max-width: $max) {
    @content;
  }
}
@mixin media-breakpoint-only() {
    @include media-breakpoint-up() {
        @include media-breakpoint-down() {
            @content;
        }
    }
}
body {
    @include media-breakpoint-only() {
        background-color: green;
    }
}
```

Result:

```
@media (min-width: 544px) {
}
@media (min-width: ) and (max-width: 768px) {
  body {
    background-color: green;
  }
}
```

Expected:

```
@media (min-width: 544px) and (max-width: 768px) {
  body {
    background-color: green;
  }
}
```

**Comment by Cerdic** [comment]
sounds similar to #409: there is a var scope issue when evaluating medias (even after the fix of #490)

**Comment by mahagr** [comment]
Thank you! I really appreciate that @Cerdic took the time to look at the issues!
